src/main/java/edu/kis/powp/jobs2d/drivers/transformators/FlippingDecorator.java

```diff
@@ -0,0 +1,59 @@
+package edu.kis.powp.jobs2d.drivers.transformators;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+public class FlippingDecorator extends TransformationsDriverDecorator {
+    public static final int FLIP_VERTICALLY = 0;
+    public static final int FLIP_HORIZONTALLY = 1;
+    public static final int FLIP_BOTH_SIDES = 2;
+
+    private int flippedX = 0, flippedY = 0;
+    private int flippingDirection;
+
+    public FlippingDecorator(Job2dDriver job2dDriver, int flippingDirection) {
```

Comment:
I think that private constructor and set of static factory methods like getFlipHorizontalDecorator(job2dDriver) would be make solution easier to use. 
flippingDirection is a name hard to crack without looking inside this class..